### PR TITLE
overload permutedims for OneElement matrices

### PR DESCRIPTION
**Pull request by max-vassili3v**
overload permutedims for OneElementMatrix to return OneElement

### CODE REVIEW

**Comment by jishnub** [comment]
This may be more general, no?

**Comment by max-vassili3v** [comment]
Implemented for more general `OneElements`, I originally excluded the implementation for `OneElementVector` since `permutedims(::AbstractVector)` uses `reshape` which keeps it as a `OneElement`

File: src/oneelement.jl [hunk 1]
```diff
@@ -427,8 +427,10 @@ function Base.reshape(A::OneElement, shape::Tuple{Vararg{Int}})
 end
 
 #permute
-
+_permute(x::Tuple, p::Tuple) = ntuple(i -> x[p[i]], length(x))
 permutedims(o::OneElementMatrix) = OneElement(o.val, reverse(o.ind), reverse(o.axes))
+permutedims(o::OneElementVector) = OneElement(o.val, (1, o.ind[1]), (1, length(o)))
```
Comment (jishnub) [inline src/oneelement.jl:432]: This should preserve the axes, I guess? The other methods do.

File: test/runtests.jl [hunk 2]
```diff
@@ -2670,6 +2670,12 @@ end
     @testset "permutedims" begin
         v = OneElement(1, (2,3), (2,5))
         @test permutedims(v) == OneElement(1, (3,2), (5,2))
+        w = OneElement(1,3,5)
+        @test permutedims(w) == OneElement(1, (1,3), (1,5))
```
Comment (jishnub) [inline test/runtests.jl:2674]: These tests should check for `===`, and not `==`.

File: src/oneelement.jl [hunk 3]
```diff
@@ -427,8 +427,10 @@ function Base.reshape(A::OneElement, shape::Tuple{Vararg{Int}})
...
+permutedims(o::OneElement, dims::Tuple) = OneElement(o.val, _permute(o.ind, dims), _permute(o.axes, dims))
```
Comment (jishnub) [inline src/oneelement.jl:433]: Do we need `dims` to be a `Tuple`? I think `AbstractVector`s would also work? Perhaps we may just have `permutedims(o::OneElement, dims)`?

**Comment by jishnub** [comment]
I don't think the method for `OneElementVector`s is necessary, as this already works on master as `reshape` is specialized for it:
```julia
julia> o = OneElement(1, 1, 2)
2-element OneElement{Int64, 1, Tuple{Int64}, Tuple{Base.OneTo{Int64}}}:
 1
 ⋅

julia> permutedims(o)
1×2 OneElement{Int64, 2, Tuple{Int64, Int64}, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}:
 1  ⋅
```

File: src/oneelement.jl (hunk 3 above)
Comment (dlfivefifty) [reply src/oneelement.jl]: Can you add a test for the case where `dims` is not a `Tuple`?

**Review by dlfivefifty — APPROVED** [review]

**Pull request merged**